### DISCUSSION TITLE
Tag pill label?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5529] [Ideas]
Asking for input before I make it an Issue request:

It would be useful if `<<tag $:/some/ugly/title>>` could display a pill with a *pretty* title.
Especially *plugins* are often served as draggable tagpills, which is a very good practice because:

* A pill is distinct and immediately recognizable for the presumptive plugin hunter who just wants to drag and drop
* It is not self evident that pills are draggable but plugin creators have an incentive to *foster* users by saying "To install the plugin just drag this pill into your TW" which is beneficial for general TW usage.
 
**The implementation** could be as simple as the tagpill showing the `caption`. Otherwise something like `<<tag "$:/ugly" "label">>`

Thoughts?

**Comment by pmario** [comment]
I'm not sure, if tag pills should be used to import plugins.

**Reply by twMat** [reply]
Why not?

**Comment by pmario** [comment]
I think there should be some "button-like" configurations. ATM I use something like:

```
<$link class="tc-btn-big-green" to="$:/config/path/to/plugin">
Drag this button to your wiki
</$link>
```
which looks like this:

![image](https://user-images.githubusercontent.com/374655/110238450-fb42e880-7f41-11eb-95b4-d5ddf2aa3730.png)

**Reply by twMat** [reply]
In what way is that better than a tag pill? I can say several things that are *not* better with it.

**Comment by Jermolene** [comment]
Hi @twMat we should avoid making the tag pill macro more complex if we can, and this seems like a different requirement. Don't forget that it's incredibly simple to make a custom draggable:

```
<$draggable filter="[tag[HelloThere]]">Drag me</$draggable>
```

**Reply by twMat** [reply]
Well, in investingating this I realized I had an old issue on a related matter... and there I realized that fellow @saqimtiaz had actually [solved this elegantly](https://github.com/Jermolene/TiddlyWiki5/issues/2843#issuecomment-720656177) with a ButtonWidget!

@Jermolene 
Off topic (but these are discussions, not Issue reports) but the DraggableWidget would be more useful if the payload for the widget could reside *between* the tags, instead of as an attribute, so that it could be anything. It seems Saq's solution referenced above uses the button as a draggable "container", but it is a bit odd that such a container would have to be a *button*. I reckon it ought to be enough with the DraggableWidget itself to drag (all) things...

**Reply by saqimtiaz** [reply]
The contents of Draggable or Button widgets are what you will visually see move on you screen when your drag. The contents are also the area you can grab and drag, i.e. the drag handle.

 Since this is more often that not different from the payload for dragging, it makes sense to have a distinct attribute for it.
